### ISSUE TITLE
 clusterversion: remove V22_2* gates (tracking issue)

### ISSUE THREAD

**Issue by celiala** [cockroachdb/cockroach#95803]
Now that we're bumping binaryMinSuportedVersion to 22.2[1], we should now remove[2] all V22_2* version gates:
- [1] https://github.com/cockroachdb/cockroach/issues/95530
	- the removing of V22_2 version gates is not blocked by 95530 (AFAIK)
- [2] as part of removing a V22_2 version gates, teams will likely also want to clean up related gated logic as applicable

This issue tracks the removal of V22_2* version gates:

- V22_2Start
- V22_2LocalTimestamps
- V22_2PebbleFormatSplitUserKeysMarkedCompacted
- V22_2EnsurePebbleFormatVersionRangeKeys
- V22_2EnablePebbleFormatVersionRangeKeys
- V22_2TrigramInvertedIndexes
- V22_2RemoveGrantPrivilege
- V22_2MVCCRangeTombstones
- V22_2UpgradeSequenceToBeReferencedByID
- V22_2SampledStmtDiagReqs
- V22_2AddSSTableTombstones
- V22_2SystemPrivilegesTable
- V22_2EnablePredicateProjectionChangefeed
- V22_2AlterSystemSQLInstancesAddLocality
- V22_2SystemExternalConnectionsTable
- V22_2AlterSystemStatementStatisticsAddIndexRecommendations
- V22_2RoleIDSequence
- V22_2AddSystemUserIDColumn
- V22_2SystemUsersIDColumnIsBackfilled
- V22_2SetSystemUsersUserIDColumnNotNull
- V22_2SQLSchemaTelemetryScheduledJobs
- V22_2SchemaChangeSupportsCreateFunction
- V22_2DeleteRequestReturnKey
- V22_2PebbleFormatPrePebblev1Marked
- V22_2RoleOptionsTableHasIDColumn
- V22_2RoleOptionsIDColumnIsBackfilled
- V22_2SetRoleOptionsUserIDColumnNotNull
- V22_2UseDelRangeInGCJob
- V22_2WaitedForDelRangeInGCJob
- V22_2RangefeedUseOneStreamPerNode
- V22_2NoNonMVCCAddSSTable
- V22_2GCHintInReplicaState
- V22_2UpdateInvalidColumnIDsInSequenceBackReferences
- V22_2TTLDistSQL
- V22_2PrioritizeSnapshots
- V22_2EnableLeaseUpgrade
- V22_2SupportAssumeRoleAuth
- V22_2FixUserfileRelatedDescriptorCorruption

Jira issue: CRDB-23749

**Comment by celiala** [comment]
Closing in favor of separate tracking issues for each team.
